### ISSUE TITLE
Response object does not handle encoding properly

### ISSUE THREAD

**Issue by harplife** [psf/requests#5680]
GET request to an API returned a response with broken encoding.
More specifically, the response body was encoded in ASCII even though the data sent via API is encoded in UTF-8.
The result is escaped non-ASCII characters in the response body.
This behaviour can be observed on r.content, r.text, and r.json().

Following the quickstart guide, I tried to "force encoding" with `r.encoding = 'UTF-8'` but it had no effect.
I made sure to include `charset=UTF-8` in request header, which also didn't help.

What ended up working was `json.loads(r.text)`.

Very simple I know, but locating the problem was difficult,
not to mention I had to be absolutely sure the API was sending a data that was encoded in UTF-8.
This problem is not mentioned or, at least, not easily found on Stackoverflow.
I thought I would mention it here, so that others can find it.

issue #5577 seems to be similar in nature, which only proves that Encoding is an issue that seems to pop up every now and then.
The answer given for that issue seems to be "there is an easy solution so there's no need to fix it on our ends".
But nonetheless, a bug is a bug.
Besides, it would be nice to mention how to troubleshoot encoding on the quickstart guide.

Sorry it's not so much of a "summary", but things had to be said.

## Expected Result

A response body properly encoded in UTF-8.
Like so, `{'body': '상품등록'}`

## Actual Result

A response body encoded in ASCII even though it's supposed to be UTF-8.
`{'body': '\uc0c1\ud488\ub4f1\ub85'}`

## Reproduction Steps

```python
import requests

url = 'https://www.sorryicannottellyoutheactualapi.com/api'
headers = {'Content-Type': 'application/json;charset=UTF-8'}
r = requests.get(url, headers=headers)
print(r.content)
print(r.text)
print(r.json())

```

## System Information

```
chardet 3.0.4
cryptography 2.8
idna 2.9
implementation CPython 3.6.7
platform Linux 4.15.0-101-generic
pyOpenSSL 19.1.0 (openssl 1010101f)
requests 2.23.0
system_ssl 1010104f
urllib3 1.25.7
using_pyopenssl True

python 3.6.7
```

**Comment by sigmavirus24** [comment]
You're right, this is a duplicate of old bug reports. The fact that you found those and opened a new one is confusing. We don't need 1 bug report for everyone who has found the same bug

**Comment by harplife** [comment]
I don't know which "old bug report" that you are referring to.
The one that I mentioned, #5577, is not the same bug, although they are similar in nature.
Perhaps there are other old bug report that I missed? If that's the case, I do apologize for not searching through the issues more thoroughly. However, it'd be nice to reference those supposed "old reports" that you mention.

Now that I look more into it, I did find another similar one, #5597, which you yourself graciously answered.
I doubt that's the old report that you are referring to though, because one would expect the behavior of `response.text` and `response.json()` to be different. And weirdly enough, they do - which can be observed when the bug I reported was fixed with `json.loads(r.text)`, and not `json.dumps(r.json())`.

I can't seem to find the 'old reports' in the open issues. Are they closed issues? I guess I assumed that the issues are closed when the bug is fixed, but I guess there are limits to some people.

Also, like I said, the bug is still there. Perhaps fix it, and document it? I figure that's what real programmers do.
